--- jquery.menu.js	2012-03-15 10:34:24.000000000 +0100
+++ JsMenu.jsni	2012-03-15 10:36:20.000000000 +0100
@@ -134,6 +113,14 @@
 				t = t.parentNode;
 			
 			//is the found node one of the visible menu elements?
+      for (k in visibleMenus) {
+        if (visibleMenus[k].$eDIV[0] == t)  {
+          // FIXME: why do we need a timeout
+          setTimeout($.Menu.closeAll, 100);
+          break;
+        }
+      }
+      // FIXME: JsQuery each doesn't work with arrays
 			if ( !$(visibleMenus).filter(function(){ return this.$eDIV[0] == t }).length )
 			{
 				$.Menu.closeAll();
@@ -239,9 +226,7 @@
 			$.extend(defaults, d);
 		},
 		prototype : {
-			/**
-			 * create / initialize new menu
-			 */
+
 			init : function()
 			{
 				var self = this;
@@ -397,7 +382,7 @@
 					this.$eDIV.css({display:'none', visibility: ''}).show();
 
 					//IEs default width: auto is bad! ie6 and ie7 have are producing different errors.. (7 = 5px shadowbox + 2px border)
-					if ( $.browser.msie )
+          if ( 0) //$.browser.msie )
 						this.$eUL.css('width', parseInt($.browser.version) == 6 ? this.$eDIV.width() - 7 : this.$eUL.width());
 
 					if ( this.settings.onOpen )
@@ -439,7 +426,7 @@
 				}
 
 				//y-pos
-				if ( $.fn.scrollTop )
+        if (0) //$.fn.scrollTop )
 				{
 					wst = $(window).scrollTop();
 					if ( wh < height ) //menu is bigger than the window
@@ -478,7 +465,7 @@
 					}
 				}
 				//x-pos
-				if ( $.fn.scrollLeft )
+        if (0) //$.fn.scrollLeft )
 				{
 					wsl = $(window).scrollLeft();
 					if ( ww + wsl < posX + width )
